components/molecules/ContributorProfileHeader/contributor-profile-header.tsx

```diff
@@ -43,6 +68,29 @@ const ContributorProfileHeader = ({
     }
   };
 
+  const handleCollaborationRequest = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (message && username) {
+      setLoading(true);
+      await requestCollaboration({ username, message });
+      setIsDialogOpen(false);
+      setLoading(false);
+      setMessage("");
+    }
+  };
+
+  const handleTextAreaInputChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
```

Comment:
We've done in this in a couple of places now. We should make this a reusable component.